leakcanary-android-proguard/src/main/java/com/squareup/leakcanary/proguard/LeakCanaryProguardExtension.kt

```diff
@@ -0,0 +1,9 @@
+package com.squareup.leakcanary.proguard
+
+open class LeakCanaryProguardExtension {
+  var obfuscatedVariantNames: Collection<String>? = null
```

Comment:
we could replace this by a `filter` method that user will have to create inside the extension. The filter would only keep debug variants by default.

Reply:
you can define the member of the extension to be a function: filter that takes a variant and returns a boolean. Users will be able to override it.

Reply:
There is a function `fun obfuscatedVariantNames(vararg names: String)` already in this class. Unless you mean something else?

Reply:
I am suggesting something different: users would define a logic / function / method / lambda. In the current model, they pass data. I am suggesting that they could define a piece of.logic to accept a variant or not.

Reply:
We were thinking what are the other options for configuration of the plugin but we couldn't figure out anything else than I implemented. Your idea sounds interesting but I don't know how would it look like in practice. Could you point me to a doc or a plugin that uses this technique?

Reply:
`var filterVariants: (BaseVariant) -> Unit = {}` inside the extension. I am not 100% sure it works, but I believe so.

Reply:
Now I get it! It indeed looks better and makes it harder to provide an incorrect variant name. I changed the extension to use this technique. Thank you.